spec.emu

```diff
@@ -21,110 +21,45 @@ contributors: Jordan Harband
-				<td>
-				</td>
-			</tr>
-			<tr>
-				<td>
-					…
-				</td>
-				<td>
-					…
-				</td>
-				<td>
-					…
-				</td>
-			</tr>
-			</tbody>
-		</table>
-	</emu-table>
-</emu-clause>
-
 <emu-clause id="sec-performpromisefinally" aoid="PerformPromiseFinally">
 	<h1>PerformPromiseFinally ( _promise_, _onFinally_, _resultCapability_ )</h1>
 	<p>The abstract operation PerformPromiseFinally performs the &ldquo;finally&rdquo; operation on _promise_ using _onFinally_ as its settlement actions. The result is _resultCapability_'s promise.</p>
 	<emu-alg>
 		1. Assert: IsPromise(_promise_) is *true*.
 		1. Assert: _resultCapability_ is a PromiseCapability Record.
 		1. If IsCallable(_onFinally_) is *false*, then
-			1. Let _onFinally_ be *undefined*.
-		1. Let _reaction_ be the PromiseReaction { [[Capability]]: _resultCapability_, [[Type]]: `"Finally"`, [[Handler]]: _onFinally_ }.
-		1. Return EnqueuePromiseReactions(_promise_, _reaction_, _reaction_, _resultCapability_).
+			1. Return PerformPromiseThen(_promise_, *undefined*, *undefined*, _resultCapability_);
+		1. Let _thenFinally_ be ! CreateThenFinally(_onFinally_).
+		1. Let _catchFinally_ be ! CreateCatchFinally(_onFinally_).
+		1. Return PerformPromiseThen(_promise_, _thenFinally_, _catchFinally_, _resultCapability_).
 	</emu-alg>
 </emu-clause>
 
-<emu-clause id="sec-performpromisethen" aoid="PerformPromiseThen">
-	<h1>PerformPromiseThen ( _promise_, _onFulfilled_, _onRejected_, _resultCapability_ )</h1>
-	<p>The abstract operation PerformPromiseThen performs the &ldquo;then&rdquo; operation on _promise_ using _onFulfilled_ and _onRejected_ as its settlement actions. The result is _resultCapability_'s promise.</p>
+<emu-clause id="sec-createthenfinally" aoid="CreateThenFinally">
+	<h1>CreateThenFinally ( _onFinally_ )</h1>
+	<p>The abstract operation CreateThenFinally takes an _onFinally_ function, and returns a callback function for use in PerformPromiseFinally.</p>
 	<emu-alg>
-		1. Assert: IsPromise(_promise_) is *true*.
-		1. Assert: _resultCapability_ is a PromiseCapability Record.
-		1. If IsCallable(_onFulfilled_) is *false*, then
-			1. Let _onFulfilled_ be *undefined*.
-		1. If IsCallable(_onRejected_) is *false*, then
-			1. Let _onRejected_ be *undefined*.
-		1. Let _fulfillReaction_ be the PromiseReaction { [[Capability]]: _resultCapability_, [[Type]]: `"Fulfill"`, [[Handler]]: _onFulfilled_ }.
-		1. Let _rejectReaction_ be the PromiseReaction { [[Capability]]: _resultCapability_, [[Type]]: `"Reject"`, [[Handler]]: _onRejected_ }.
-		1. Return EnqueuePromiseReactions(_promise_, _fulfillReaction_, _rejectReaction_, _resultCapability_).
+		1. Assert: IsCallable(_onFinally_) is *true*.
+		1. Return a function that takes one argument, _value_, and when invoked, performs the following steps:
+			1. Let _result_ be ? Call(_onFinally_, *undefined*).
+			1. Let _promise_ be ! PromiseResolve(%Promise%, _result_);
+			1. Let _valueThunk_ be equivalent to a function that returns _value_.
```

Comment:
No, that explanatory table is not normative; the meaning column is a non-normative description (and ideally should be marked as such). You can tell because if you changed to it to read "floop florp florble" no parts of the implementation would change.

Anyway, there's a big difference here, which is that the valueThunk closes over value. It's not just processing its argument. And to close over something, you need to go through the whole process of creating a subsection and declaring what internal slots the function has to represent closed-over values.